Tribler/Policies/BoostingSource.py

```diff
@@ -0,0 +1,492 @@
+
+        self.unavail_torrent = {}
+
+    def kill_tasks(self):
+        BoostingSource.kill_tasks(self)
+
+        self.session.remove_observer(self._on_database_updated)
+
+    def _load(self, dispersy_cid):
+        dispersy = self.session.get_dispersy_instance()
+
+        @blocking_call_on_reactor_thread
+        def join_community():
+            """
+            find the community/channel id, then join
+            """
+            try:
+                self.community = dispersy.get_community(dispersy_cid, True)
+                self.boosting_manager.register_task(str(self.source) + "_get_id",
+                                                    reactor.callLater(1, get_channel_id))
+
+            except CommunityNotFoundException:
+
+                allchannelcommunity = None
+                for community in dispersy.get_communities():
+                    if isinstance(community, AllChannelCommunity):
+                        allchannelcommunity = community
+                        break
+
+                if allchannelcommunity:
+                    self.community = ChannelCommunity.init_community(dispersy, dispersy.get_member(mid=dispersy_cid),
+                                                                     allchannelcommunity._my_member, self.session)
+                    self._logger.info("Joined channel community %s", dispersy_cid.encode("HEX"))
+                    self.boosting_manager.register_task(str(self.source) + "_get_id",
+                                                        reactor.callLater(1, get_channel_id))
+                else:
+                    self._logger.error("Could not find AllChannelCommunity")
+
+        def get_channel_id():
+            """
+            find channel id by looking at the network
+            """
+            if self.community and self.community._channel_id:
+                self.channel_id = self.community._channel_id
+
+                channel_dict = self.boosting_manager.channelcast_db.getChannel(self.channel_id)
+                self.channel = Channel(*channel_dict)
+
+                if not self.boosting_manager.is_pending_task_active(str(self.source) + "_update"):
+                    self._logger.debug("Registering update call")
+                    self.boosting_manager.register_task(str(self.source) + "_update", LoopingCall(self._update)).start(
+                        self.interval, now=True)
+                self._logger.info("Got channel id %s", self.channel_id)
+
+                self.ready = True
+            else:
+                self._logger.warning("Could not get channel id, retrying in 10 s")
+                self.boosting_manager.register_task(str(self.source) + "_get_id",
+                                                    reactor.callLater(10, get_channel_id))
+
+        self.boosting_manager.register_task(str(self.source) + "_join_comm",
+                                            reactor.callLater(1, join_community))
```

Comment:
If you are calling join_community like this then you don't need the blocking call on thread decorator as it is already being called there.